iothub/device/src/DirectMethod/DirectMethodRequest.cs

```diff
@@ -84,7 +84,12 @@ public DirectMethodRequest(string methodName)
         /// <summary>
         /// The direct method payload.
         /// </summary>
-        protected internal byte[] Payload { get; set; }
+        /// <remarks>
+        /// The direct method request payload is to be set by the client application only when using <see cref="IotHubModuleClient"/>
+        /// to invoke a direct method on an edge device or on an edge module.
+        /// </remarks>
+        [JsonProperty("payload", NullValueHandling = NullValueHandling.Include)]
+        public byte[] Payload { get; set; }
```

Comment:
As a user, I would like the payload here to be settable the same way it is set on the service client ([as type](https://github.com/Azure/azure-iot-sdk-csharp/blob/36883b515d41dadd85d733e7b04abdd95bbcfce9/iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs#L36) ```object```, currently). Is there some reason why this payload is different?

Reply:
And what would happen if the user passes in gibberish bytes here?

Reply:
And it's fine if this is still to be worked out. If all we want to do for now is make this public, then I have no objections.

Reply:
> As a user, I would like the payload here to be settable the same way it is set on the service client ([as type](https://github.com/Azure/azure-iot-sdk-csharp/blob/36883b515d41dadd85d733e7b04abdd95bbcfce9/iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs#L36) `object`, currently). Is there some reason why this payload is different?

I don't have any argument in favor of making this bytes. This was simply the path of minimal change. I can see the value in making this similar to service client's direct method request.

Reply:
> And what would happen if the user passes in gibberish bytes here?

The SDK doesn't validate any payload. It would be on service to validate this based on their defined contract.

Reply:
> The SDK doesn't validate any payload. It would be on service to validate this based on their defined contract.

I forgot that the device side of this just sends the payload as-is (whereas the service client nests it in a JSON object and then serializes it all). This may actually be the best form for the module client's payload to take since it involves no SDK serialization so users get the most flexibility

Reply:
I'd rather it be settable via ctor. What prevents that?

Reply:
Adding a new class for edge module direct methods